rust/osm2lanes/src/lib.rs

```diff
@@ -4,6 +4,8 @@
 //! WARNING: The output specification and all of this code is just being prototyped. Don't depend
 //! on anything yet.
 
+#![feature(is_some_with)]
```

Comment:
This would require nightly rust to build, right? Since we're already requiring it for formatting, should we just make use of it everywhere?

(I'd push back a little -- I think it would require all downstream consumers to use nightly too. It's a pity, since there are features like this that're quite handy)

Reply:
why nightly?

Reply:
``#![feature]` may not be used on the stable release channel`
https://github.com/rust-lang/rust/issues/93050
Until this is stabilized, I think this forces us to use nightly rust

Reply:
it seems features like that are unlikely to ever land in stable, I am actually surprised that one went in:
https://github.com/rust-lang/rust/pull/75298
we can go through the cases where it was used here and replace it with something slightly more verbose :)

Reply:
I agree with staying on stable. I didn't know what `#![feature]` was, but IntelliJ suggested it, so I thought I'd get your input on it. Now I know!

I only used it one time, and can easily work around it.